.github/workflows/docker-build-static.yml

```diff
@@ -151,82 +112,90 @@ jobs:
-          path: target/aarch64-unknown-linux-musl/release/migration-tool
+          name: ${{ matrix.target }}-migration-tool-bin
+          path: target/${{ matrix.target }}/release/migration-tool
 
-  build-amd64:
+  lldap-test:
+    needs: [build-ui,build-bin]
+    name: LLDAP test
     runs-on: ubuntu-latest
-    container:
-      image: nitnelave/rust-dev:latest
-      env:
-        CARGO_TERM_COLOR: always
-        RUSTFLAGS: -Ctarget-feature=+crt-static
-        CARGO_HOME: ${GITHUB_WORKSPACE}/.cargo
-        CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER: x86_64-linux-musl-gcc
+    services:
+        mariadb:
+          image: mariadb:latest
+          ports:
+            - 3306:3306
+          env:
+            MYSQL_USER: lldapuser
+            MYSQL_PASSWORD: lldappass
+            MYSQL_DATABASE: lldap
+            MYSQL_ROOT_PASSWORD: rootpass
+
+        postgresql:
+           image: postgres:latest
+           ports:
+             - 5432:5432
+           env:
+             POSTGRES_USER: lldapuser
+             POSTGRES_PASSWORD: lldappass
+             POSTGRES_DB: lldap
+
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3.3.0
-      - uses: actions/cache@v3
-        with:
-          path: |
-            .cargo/bin
-            .cargo/registry/index
-            .cargo/registry/cache
-            .cargo/git/db
-            target
-          key: lldap-bin-amd64-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            lldap-bin-amd64-
-      - name: install musl
-        run: apt update && apt install -y musl-tools tar wget
-#      - name: fetch musl-gcc
-#        run: |
-#             wget -c https://musl.cc/x86_64-linux-musl-cross.tgz
-#             tar zxf ./x86_64-linux-musl-cross.tgz -C /opt
-#             echo "/opt/x86_64-linux-musl-cross:/opt/x86_64-linux-musl-cross/bin" >> $GITHUB_PATH
-      - name: add x86_64 target
-        run: rustup target add x86_64-unknown-linux-musl
-      - name: build x86_64 lldap
-        run: cargo build --target=x86_64-unknown-linux-musl --release -p lldap -p migration-tool
-      - name: check path
-        run: ls -al target/x86_64-unknown-linux-musl/release/
-      - name: upload amd64 lldap artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: amd64-lldap-bin
-          path: target/x86_64-unknown-linux-musl/release/lldap
-      - name: upload amd64 migration-tool artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: amd64-migration-tool-bin
-          path: target/x86_64-unknown-linux-musl/release/migration-tool
+       - name: Download All Artifacts
+         uses: actions/download-artifact@v3
+         with:
+           path: bin/
+       - name: Set executables
+         run: chmod +x bin/*/lldap
+       - name: run lldap with postgres DB
+         run: |
+              bin/x86_64-unknown-linux-musl-lldap-bin/lldap run &
+              sleep 10s
```

Comment:
Ha! That's one way to do it :D 
Does that actually work?

Reply:
If so, we should also do one with sqlite.

Reply:
Well it works, and `lldap selfcheck` pretty much success. Idk if it enough, it just ensure it run fine first. ☕

Reply:
LLDAP Healthcheck doesn't actually return a non-0 code if there are errors, though. You'd need to grep for ERROR in the output (or I can make a change).

Reply:
Fixing it in https://github.com/nitnelave/lldap/pull/421